control-plane/connect-inject/webhook_v2/mesh_webhook.go

```diff
@@ -0,0 +1,572 @@
+		return false, nil
+	}
+
+	// If we already injected then don't inject again
+	if pod.Annotations[constants.KeyMeshInjectStatus] != "" || pod.Annotations[constants.KeyInjectStatus] != "" {
+		return false, nil
+	}
+
+	// If the explicit true/false is on, then take that value. Note that
+	// this has to be the last check since it sets a default value after
+	// all other checks.
+	if raw, ok := pod.Annotations[constants.AnnotationMeshInject]; ok {
+		return strconv.ParseBool(raw)
+	}
+
+	return !w.RequireAnnotation, nil
+}
+
+func (w *MeshWebhook) defaultAnnotations(pod *corev1.Pod, podJson string) error {
+	if pod.Annotations == nil {
+		pod.Annotations = make(map[string]string)
+	}
+
+	pod.Annotations[constants.AnnotationOriginalPod] = podJson
+	pod.Annotations[constants.AnnotationConsulK8sVersion] = version.GetHumanVersion()
+
+	return nil
+}
+
+// prometheusAnnotations sets the Prometheus scraping configuration
+// annotations on the Pod.
+func (w *MeshWebhook) prometheusAnnotations(pod *corev1.Pod) error {
+	enableMetrics, err := w.MetricsConfig.EnableMetrics(*pod)
+	if err != nil {
+		return err
+	}
+	prometheusScrapePort, err := w.MetricsConfig.PrometheusScrapePort(*pod)
+	if err != nil {
+		return err
+	}
+	prometheusScrapePath := w.MetricsConfig.PrometheusScrapePath(*pod)
+
+	if enableMetrics {
+		pod.Annotations[constants.AnnotationPrometheusScrape] = "true"
+		pod.Annotations[constants.AnnotationPrometheusPort] = prometheusScrapePort
+		pod.Annotations[constants.AnnotationPrometheusPath] = prometheusScrapePath
+	}
+	return nil
+}
+
+// consulNamespace returns the namespace that a service should be
+// registered in based on the namespace options. It returns an
+// empty string if namespaces aren't enabled.
+func (w *MeshWebhook) consulNamespace(ns string) string {
+	return namespaces.ConsulNamespace(ns, w.EnableNamespaces, w.ConsulDestinationNamespace, w.EnableK8SNSMirroring, w.K8SNSMirroringPrefix)
+}
+
+func findServiceAccountVolumeMount(pod corev1.Pod) (corev1.VolumeMount, string, error) {
+	// Find the volume mount that is mounted at the known
+	// service account token location
+	var volumeMount corev1.VolumeMount
+	for _, container := range pod.Spec.Containers {
+		for _, vm := range container.VolumeMounts {
+			if vm.MountPath == "/var/run/secrets/kubernetes.io/serviceaccount" {
+				volumeMount = vm
+				break
+			}
+		}
+	}
+
+	// Return an error if volumeMount is still empty
+	if (corev1.VolumeMount{}) == volumeMount {
+		return volumeMount, "", errors.New("unable to find service account token volumeMount")
+	}
+
+	return volumeMount, "/var/run/secrets/kubernetes.io/serviceaccount/token", nil
+}
+
+func (w *MeshWebhook) annotatedServiceNames(pod corev1.Pod) []string {
+	var annotatedSvcNames []string
+	if anno, ok := pod.Annotations[constants.AnnotationService]; ok {
+		annotatedSvcNames = strings.Split(anno, ",")
+	}
+	return annotatedSvcNames
+}
```

Comment:
Is this used anywhere now? I think we can deprecate it since we decouple the services.